### ISSUE TITLE
[CNN] 이거 트레인 어큐러시는 100이고 테스트만 좀 올려보자ㅎㅎㅎㅎ

### ISSUE THREAD

**Issue by ghost** [PARKMIHE/AICOCO#29]
from google.colab import drive
drive.mount('/content/gdrive')

**Comment by ghost** [comment]
import random
import numpy as np
import tensorflow as tf
import cv2
import os

from random import shuffle
from tqdm import tqdm

train_dir = '/content/gdrive/My Drive/Colab Notebooks/flowers'
test_dir = '/content/gdrive/My Drive/Colab Notebooks/flowers'

img_size = 28

def label_folder(folders):
  
    if folders == 'forsythia': return [1,0,0,0,0,0,0,0,0,0]
    elif folders == 'buckwheat': return [0,1,0,0,0,0,0,0,0,0]
    elif folders == 'sunflower': return [0,0,1,0,0,0,0,0,0,0]
    elif folders == 'cosmos': return [0,0,0,1,0,0,0,0,0,0]
    elif folders == 'lily': return [0,0,0,0,1,0,0,0,0,0]
    elif folders == 'roseofsharon': return [0,0,0,0,0,1,0,0,0,0]
    elif folders == 'tulip': return [0,0,0,0,0,0,1,0,0,0]
    elif folders == 'cherryblossom': return [0,0,0,0,0,0,0,1,0,0]       #메밀꽃
    elif folders == 'rose': return [0,0,0,0,0,0,0,0,1,0]       #진달래
    elif folders == 'korearose': return [0,0,0,0,0,0,0,0,0,1]   #
def create_train_data():
    training_data = []
    dirs = os.listdir( train_dir )
    for folders in dirs:
        label = label_folder(folders)
        req_train_dir = os.path.join(train_dir,folders)
        for img in tqdm(os.listdir(req_train_dir)):
            path = os.path.join(req_train_dir,img)
            img = cv2.imread(path,cv2.IMREAD_GRAYSCALE)
            img = cv2.resize(img, (img_size,img_size))
            training_data.append([np.array(img),np.array(label)])
    shuffle(training_data)
    np.save('flower_train_data.npy', training_data)
    return training_data

def process_test_data():
    testing_data = []
    for img in tqdm(os.listdir(test_dir)):
        path = os.path.join(test_dir,img)
        img_num = img.split('.')[0]  #image id
        img = cv2.imread(path,cv2.IMREAD_GRAYSCALE)
        testing_data.append([np.array(img), img_num])
        
    shuffle(testing_data)
    np.save('flower_test_data.npy', testing_data)
    return testing_data

train_data = create_train_data()

**Comment by ghost** [comment]
# Convolutional Layer 1.
filter_size1 = 4 
num_filters1 = 32
# Convolutional Layer 2.
filter_size2 = 4
num_filters2 = 64
# Convolutional Layer 3.
filter_size3 = 4
num_filters3 = 128
# Convolutional Layer 4
filter_size4 = 4
num_filters4 = 256
# Convolutional Layer 5
filter_size5 = 4
num_filters5 = 128
# Fully-connected layer.
fc_size = 1024         

num_channels = 1
# image dimensions (only squares for now)   #################
img_size = 28
# Size of image when flattened to a single dimension
img_size_flat = img_size * img_size * num_channels
# Tuple with height and width of images used to reshape arrays.
img_shape = (img_size, img_size)
# class 써줌
classes = ['forsythia', 'buckwheat', 'sunflower', 'cosmos', 'lily','roseofsharon','tulip','cherryblossom','rose','korearose']
num_classes = len(classes)
# batch size
batch_size = 50
keep_prob=tf.placeholder(tf.float32)


print(len(train_data))
print(len(test_data))
#data 불러옴
checkpoint_dir = "models/"

#train_data = np.load('flower_train_data.npy')

def new_weights(shape):
    return tf.Variable(tf.truncated_normal(shape, stddev=0.05))
def new_biases(length):
    return tf.Variable(tf.constant(0.05, shape=[length]))

def new_conv_layer(input,num_input_channels,filter_size,num_filters,use_pooling=True):  

    shape = [filter_size, filter_size, num_input_channels, num_filters]
    weights = new_weights(shape=shape)
    biases = new_biases(length=num_filters)

    layer = tf.nn.conv2d(input=input,filter=weights,strides=[1, 1, 1, 1],padding='SAME')
    layer += biases

    if use_pooling:
        layer = tf.nn.max_pool(value=layer,ksize=[1, 2, 2, 1],strides=[1, 2, 2, 1],padding='SAME')

    layer = tf.nn.relu(layer)
    
    return layer, weights

def new_fc_layer(input,num_inputs,num_outputs,use_relu=True): 

    weights = new_weights(shape=[num_inputs, num_outputs])
    biases = new_biases(length=num_outputs)

    layer = tf.matmul(input, weights) + biases
    
    if use_relu:
        layer = tf.nn.relu(layer)

    return layer

def flatten_layer(layer):
    
    layer_shape = layer.get_shape()
    num_features = layer_shape[1:4].num_elements()
    layer_flat = tf.reshape(layer, [-1, num_features])

    return layer_flat, num_features

x = tf.placeholder(tf.float32, shape=[None, img_size_flat], name='x')
x_image = tf.reshape(x, [-1, img_size, img_size, num_channels])
y_true = tf.placeholder(tf.float32, shape=[None, num_classes], name='y_true')
y_true_cls = tf.argmax(y_true, axis=1)

layer_conv1, weights_conv1 = new_conv_layer(input=x_image,num_input_channels=num_channels,filter_size=filter_size1,
                                            num_filters=num_filters1,use_pooling=True)
layer_conv1=tf.nn.dropout(layer_conv1,keep_prob=keep_prob)
layer_conv2, weights_conv2 = new_conv_layer(input=layer_conv1,num_input_channels=num_filters1,filter_size=filter_size2,
                                            num_filters=num_filters2,use_pooling=True)
layer_conv2=tf.nn.dropout(layer_conv2,keep_prob=keep_prob)
layer_conv3, weights_conv3 = new_conv_layer(input=layer_conv2,num_input_channels=num_filters2,filter_size=filter_size3,
                                            num_filters=num_filters3,use_pooling=True)
layer_conv3=tf.nn.dropout(layer_conv3,keep_prob=keep_prob)
layer_conv4, weights_conv4 = new_conv_layer(input=layer_conv3,num_input_channels=num_filters3,filter_size=filter_size4,
                                            num_filters=num_filters4,use_pooling=True)
layer_conv5=tf.nn.dropout(layer_conv4,keep_prob=keep_prob)
layer_conv5, weights_conv5 = new_conv_layer(input=layer_conv4,num_input_channels=num_filters4,filter_size=filter_size5,
                                            num_filters=num_filters5,use_pooling=True)
layer_conv5=tf.nn.dropout(layer_conv5,keep_prob=keep_prob)
layer_flat, num_features = flatten_layer(layer_conv5)
layer_fc1 = new_fc_layer(input=layer_flat,num_inputs=num_features,num_outputs=fc_size,use_relu=True)
layer_fc2 = new_fc_layer(input=layer_fc1,num_inputs=fc_size,num_outputs=num_classes,use_relu=False)

y_pred = tf.nn.softmax(layer_fc2)
y_pred_cls = tf.argmax(y_pred, axis=1)
cross_entropy = tf.nn.softmax_cross_entropy_with_logits(logits=layer_fc2,labels=y_true)
cost = tf.reduce_mean(cross_entropy)

optimizer = tf.train.AdamOptimizer(learning_rate=1e-4).minimize(cost)
correct_prediction = tf.equal(y_pred_cls, y_true_cls)
accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
#train,test 비율은 드라이브에 있는 전체 사진 갯수 기준으로 7:3으로 넣어주면돼 이건 20000장 기준!
train = train_data[:-14000]
test = train_data[-6000:]

x_batch = np.array([i[0] for i in train]).reshape(-1,img_size_flat)
#x_batch = np.array([i[0] for i in train])

y_true_batch = [i[1] for i in train]

test_x = np.array([i[0] for i in test]).reshape(-1,img_size_flat)
#test_x = np.array([i[0] for i in test])

test_y = [i[1] for i in test]
print(len(test_x),len(test_y))
session = tf.Session()
session.run(tf.global_variables_initializer())
total_iterations = 0

def optimize(num_iterations):
    
    global total_iterations

    for i in range(total_iterations,total_iterations + num_iterations):
      
      a = 0
      for __ in range(int(len(train)/batch_size)):
        
        
          feed_dict_train = {x: x_batch[a:a+batch_size,:],y_true: y_true_batch[a:a+batch_size],keep_prob:1.0}
          session.run(optimizer, feed_dict=feed_dict_train)
          a = a + batch_size
       
      if i % 2 == 0:
         print("Iteration = ", i, "Loss = ", session.run(cost, feed_dict=feed_dict_train),
                      "Train Accuracy = ", session.run(accuracy, feed_dict=feed_dict_train), 
                      "Test Accuracy = ", session.run(accuracy, feed_dict={x:test_x, y_true: test_y,keep_prob:1.0}))

optimize(num_iterations=50)

**Comment by ghost** [comment]
저거 위에서부터 한 블록씩 한 코드 셀에 실행하면 된다 안되거나 모르겠으면 말해줘!
